agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/heartbeat/HeartbeatExporter.java

```diff
@@ -125,7 +127,9 @@ private void send() {
       telemetryItemsConsumer.accept(Collections.singletonList(gatherData()));
       logger.trace("No of heartbeats sent, {}", ++heartbeatsSent);
     } catch (RuntimeException e) {
-      logger.warn("Error occured while sending heartbeat");
+      try (MDC.MDCCloseable ignored = AzureMonitorMsgId.HEARTBEAT_SEND_ERROR.makeActive()) {
+        logger.warn("Error occured while sending heartbeat");
+      }
```

Comment:
another one where I'm not sure the purpose of the catch block, I don't think we ever throw exception from accept above(?)